### ISSUE TITLE
Need to reinitialize enabled currencies after update

### ISSUE THREAD

**Issue by ismaeldcom** [Automattic/woocommerce-payments#2244]
Working on #2243 I realize that we don't need to use `invalidateResolutionForStoreSelector`. This was used to force a refresh of enabled currencies list. But also show an empty list while it's loading.

The problem was that when we make an update request on the client, we're updating the option and returning the **old** enabled currencies, After a quick patch in https://github.com/Automattic/woocommerce-payments/pull/2243/commits/8532e7ae8f7bcace82d1058cc48adb9c33e51c0d It works as expected, returning the updated enabled currencies and allowing a better UX.

---

After an internal conversation, we decide to implement a better solution after we merge #2243. The general idea is to be clearer about calling `initialize_enabled_currencies`. So one solution might be to have a new method `reinitialize_enabled_currencies` that will remove `enabled_currencies` call `initialize_enabled_currencies` again. And call it from `set_enabled_currencies`. And be sure that `get_enabled_currencies` didn't return an outdated value.

We should also add a test case for this, to prevent any future regression.

**Comment by jessepearson** [comment]
> one solution might be to have a new method reinitialize_enabled_currencies that will remove enabled_currencies call initialize_enabled_currencies again. And call it from set_enabled_currencies. 

I just looked at the code, and this is mostly what already happens. `$this->enabled_currencies` is (re)set to an empty array in `initialize_enabled_currencies`, so that value doesn't need to be reset before that. I'm thinking that if we want to have a `reinitialize_enabled_currencies` method to make it more clear about what's happening, we can create it just to call `initialize_enabled_currencies`. Not sure if that's needed, though.

**Comment by dmallory42** [comment]
Me and @jessepearson just had a sync chat about this issue and we feel that the solution @ismaeldcom implemented is sufficient for now as it solves the problem and ensures that `set_enabled_currencies` updated the `$enabled_currencies` variable as well as the option value and that `get_enabled_currencies` always returns the most up to date version of the currencies. 

Therefore we decided to close this issue but consider whether we need to do any refactors of the Multi Currency class after the MVP has shipped. I've opened an issue https://github.com/Automattic/woocommerce-payments/issues/2378 to track this.
